src/ui/components/SubmitProposal.tsx

```diff
@@ -76,19 +76,18 @@ export function SubmitProposal() {
         />
         <Padder scale={1} />
 
-        <Paragraph>{`Locked balance:`}</Paragraph>
-        <TextInput
-          dense
-          autoComplete="off"
-          mode="outlined"
-          placeholder="Place your balance"
-          keyboardType="decimal-pad"
-          value={state.balance}
-          onChangeText={(nextValue) => dispatch({type: 'SET_BALANCE', payload: decimalKeypad(nextValue)})}
-          contextMenuHidden={true}
-          right={<TextInput.Affix text={(api && formatBalance(getBalanceFromString(api, state.balance))) ?? ''} />}
-        />
-        <MaxBalance address={state.account} />
+        {api && state.account && (
```

Comment:
According to this logic the `BalanceInput` component wont show until an account is selected. We can pass the api and account as optional props to the component and handle it there.